csharp/extractor/Semmle.Extraction.CSharp/Entities/Parameter.cs

```diff
@@ -112,6 +112,13 @@ public override void Populate(TextWriter trapFile)
             foreach (var l in symbol.Locations)
                 trapFile.param_location(this, Context.Create(l));
 
+            if (!symbol.Locations.Any() &&
+                symbol.ContainingSymbol is IMethodSymbol ms &&
+                ms.Name == WellKnownMemberNames.TopLevelStatementsEntryPointMethodName)
```

Comment:
We should validate that the containing type is also the well known `<Program>$`.